### ISSUE TITLE
NfaUtil#findCycles compile error

### ISSUE THREAD

**Issue by kthoms** [eclipse/xtext-core#212]
In my devenv I am experiencing the following compile error:
```
Description	Resource	Path	Location	Type
Cannot infer type argument(s) for <S> findCycles(Nfa<S>, S, IAcceptor<List<S>>, Map<S,Integer>, LinkedList<S>)	NfaUtil.java	/org.eclipse.xtext.util/src/org/eclipse/xtext/util/formallang	line 601	Java Problem
```

![screenshot 17](https://cloud.githubusercontent.com/assets/265597/21457413/7c4cae3e-c92f-11e6-91ec-99a775a51512.png)

**Comment by kthoms** [comment]
I think this is a JDT bug. I upgraded today from 3.12.1 to 3.12.2 and did not have thid issue before. Opened https://bugs.eclipse.org/bugs/show_bug.cgi?id=509694

**Comment by kthoms** [comment]
It is a confirmed JDT regression. I think we should rely on a fix and rewrite the method in a way compatible with JDT 3.12.2.
@meysholdt Could you take are of it since you wrote the original implementation?

**Comment by kthoms** [comment]
Scheduled fix for 4.6.3, see https://bugs.eclipse.org/bugs/show_bug.cgi?id=508834

**Comment by meysholdt** [comment]
if passing explicit type params to the invocation of "findCycles()" does the trick, I'm fine with it.

**Comment by kthoms** [comment]
Fixed via f8ccfdc68908773f318641e6fcad74a3afd46ab4
